app/views/datasets/dataset.html.erb

```diff
@@ -22,7 +22,8 @@
 		</div>
   <% end %>
 
-  <%= form_for @dataset, action: 'create', method: 'post' do |f| %>
+  <%= form_for @dataset, url: editing? ? dataset_path(@dataset) : datasets_path,
+                         method: update_method do |f| %>
```

Comment:
Something funny around here. If you try to submit the form without a required field the url changes to `/datasets` which means you can't then submit a post request.